spec/service_providers/pact_ruby_fetch_pacts_test.rb

```diff
@@ -0,0 +1,292 @@
+              },
+              'pb:latest-provider-pacts-with-tag' => {
+                'href' => 'http://localhost:1234/pacts/provider/{provider}/latest/{tag}',
+                'title' => 'Latest pacts by provider with the specified tag'
+              },
+              :'pb:provider-pacts-with-tag' => {
+                'href' => 'http://localhost:1234/pacts/provider/{provider}/tag/{tag}',
+                'title' => 'All pact versions for the provider with the specified consumer version tag'
+              },
+              'pb:provider-pacts' => {
+                'href' => 'http://localhost:1234/pacts/provider/{provider}',
+                'title' => 'All pact versions for the specified provider'
+              }
+            }
+          }
+        )
+    end
+
+    context 'retrieving latest pacts by provider' do
+      let(:tags) {nil}
+      let(:all_pacts) {false}
+
+      before do
+        pact_broker
+          .given('pacts exist between consumer and provider')
+          .upon_receiving('a request to retrieve the latest pacts for provider')
+          .with(
+            method: :get,
+            path: '/pacts/provider/provider-name/latest',
+            headers: get_headers
+          ).
+          will_respond_with(
+            status: 200,
+            headers: {},
+            body: {
+              _links: {
+                self: {
+                  href: 'http://localhost:1234/pacts/provider/provider-name/latest',
+                  title: 'Latest pact versions for the provider'},
+                provider: {
+                  href: 'http://localhost:1234/pacticipants/provider-name',
+                  'title': 'provider-name'
+                },
+                pacts: [
+                  {'href': 'pact-brker-url-for-consumer-1'
+                  },
+                  {'href': 'pact-broker-url-for-consumer-2'
+                  }
+                ]
+              }
+            }
+          )
+      end
+
+      it 'returns the array of pact urls' do
+        pacts = Pact::PactBroker::FetchPacts.call(provider, tags, broker_base_url, basic_auth_options, all_pacts)
+
+        expect(pacts).to eq(%w(pact-brker-url-for-consumer-1 pact-broker-url-for-consumer-2))
+      end
+    end
+
+    context 'retrieving latest pacts by provider with the specified tag' do
+      let(:tags) {%w[tag-1 tag-2]}
+      let(:all_pacts) {false}
+
+      before do
+        pact_broker
+          .given('pacts exist between consumer and provider')
+          .upon_receiving('a request to retrieve the latest tagged (tag-1) pacts for provider')
+          .with(
+            method: :get,
+            path: '/pacts/provider/provider-name/latest/tag-1',
+            headers: get_headers
+          ).
+          will_respond_with(
+            status: 200,
+            headers: {},
+            body: {
+              _links: {
+                self: {
+                  href: 'http://localhost:1234/pacts/provider/provider-name/latest/tag-1',
```

Comment:
Someone reading this url might not be sure if "provider-name" was the parameter name or the parameter value.